### ISSUE TITLE
Admin users and User Administration screen

### ISSUE THREAD

**Issue by frankduncan** [redcross/smoke-alarm-portal#263]
Currently creating and modifying users is done by hand from the database.  This should be updated so that admins of the system can do this through the UI after logging in.

Changes needed:
* Admin flag added
* User add/update screens added and enabled for admins
* Able to set regions to users
* ~~Able to make users the contact for regions~~ (moved to its own issue #264)

**Comment by frankduncan** [comment]
Related to #138

**Comment by kfogel** [comment]
UI/UX feedback:

* Regarding the columns on admin/users page
  - Regarding "Add New" box in upper right
    + This is where people would normally expect Search.
      (I realize the positioning may be due to drywall and is not trivial to change; we can leave it in that position if so.)
    + Can we move "Add New" to left side of input box?
    + Change the button to say "Add New User: ", for clarity.
      (Or if we don't move it to left side of input box, then omit the ": " in that.)
    + Change gray "enter a username" text to "enter new username".
  - Search box
    + Search box should have gray "search users" text in it.
    + Search bx show magnifying glass?
  - Column headers
    + Can there be tooltips for the column headers?
      It's not immediately obvious what "siteAdmin" and "active" mean.
    + Let's not use internal representations for display.  Instead of "siteAdmin", say "Is site admin?".  Instead of "active", say "Is active?"  (The tooltip can then give a more detailed explanation.)
    + Can the "active" indicator be a checkbox instead of yes/no?  That way one can be deactivated directly from this panel, instead of having to click "Edit" on that user.
    + Same with site admin :-).
* Creating a New User
  I entered "fish" as the username and hit "Add New".  I didn't expect that that would create the user before I had entered any information about the user!  But it turns out it did.  I hit the Back Button immediately to go back to the admin/users page; the new "fish" user wasn't showing yet, but that's only because the page hadn't refreshed (because I got there via Back Button).  Once I reloaded the page, "fish" was listed.  
* On the New User page
   - On this page, instead of "Is Active" being a checkbox, make it choice?  Make active/inactive radio button, and default to active.
   - What does "Update Regions" header title mean?
   - For the "Update Regions" section title, include the username: "Update regions for user FOO" (because the username is otherwise not visible anywhere on the screen by that point, and the person should always be able to look at the screen and know what they're doing).
   - In the Update Regions section: Instead of "View" say "Can view region requests", and instead of "Contact" say "Set user as contact for region".  Use tooltips to explain in more detail what each one does.  And make both labels column headers, instead of repeating them on each region row.
   - The "No region found" region is showing as a region.  That's confusing.
   - In the "Danger Zone" at the bottom, get rid of "If you do this, it cannot be undone. You may also create orphaned document relationships too."  In the warning box that pops up, we should say the username being deleted, say that it cannot be undone, and explain what's actually going on with requests if that user is deleted ("orphaned document relationships" doesn't mean anything to Red Cross volunteer admins).
   - After you're done and you hit Update, it should go back to the admin/users page.
   - Relatedly, there should be a link back to admin/users page too.
   - If you click Delete, a confirmation box pops up and asks "Are you sure?" but doesn't include the username in the question.  It should.

Also, a more general bug related the New User page: 

I created "testy" and immediately hit the Back Button to go back to the admin/users page.  As expected, "testy" is now a user, with no fields set, and is active but not a site admin.  Fine.  I click "Edit" for "testy", and am brought back to the user's page (same as it looked when I did New User).  I check the "site admin" checkbox, and now... what do I do to submit just that change?  I can't hit the "Update" button, b/c there's no email address and email is required.

Basically, the process of creating a user is a little wonky right now.  Once you enter the username and hit New User, the database is now in a state it's not "supposed" to be in: the user already exists, but doesn't have some of the required information.  We shouldn't permit that.  Let's not create the user until we have all the required information.

**Comment by OhMcGoo** [comment]
@kfogel, @frankduncan: The user administration features look good. I was able to create a user with any issues. I assume all the existing users will populate the **Users** screen when these features are deployed. One small request: Can you put **Administration: Users** above **Reports: Smoke Alarm Installation Requests** on the getasmokealarm.org/admin/ page?

**Comment by frankduncan** [comment]
@OhMcGoo (cc @kfogel) The reordering is done and on demo :)

**Comment by frankduncan** [comment]
With pull #280 hitting production, I'm closing this back out.
